lib/Cake/Test/Case/Controller/Component/CookieComponentTest.php

```diff
@@ -790,9 +752,99 @@ public function testDeleteChildrenNotExist() {
 	}
 
 /**
+ * Test that deleting a child entry of the cookie array works
+ *
+ * @return void
+ */
+	public function testDeleteChildEntry() {
+		$response = $this->Controller->response;
+
+		$this->Cookie->write('User.name', 'user', false);
+		$this->Cookie->write('User.email', 'user@example.com', false);
+
+		// Emulate the next request from the client.
+		$cookies = $response->cookie();
+		$_COOKIE = $this->_rebuildCookie($cookies);
+		$this->Cookie->startup($this->Controller);
+
+		$this->assertEquals('user', $this->Cookie->read('User.name'));
+		$this->assertEquals('user@example.com', $this->Cookie->read('User.email'));
+
+		// Delete a child entry
+		$this->Cookie->delete('User.name');
+
+		// Emulate another request from the client.
+		$cookies = $response->cookie() + $cookies; // The previous cookie should be merged.
+		$_COOKIE = $this->_rebuildCookie($cookies);
+		$this->Cookie->startup($this->Controller);
+
+		// The client shouldn't send the deleted entry
+		$this->assertNull($this->Cookie->read('User.name'));
+		$this->assertEquals('user@example.com', $this->Cookie->read('User.email'));
+	}
+
+/**
+ * Test that writing cookies with differnet expiration times works
+ *
+ * @return void
+ */
+	public function testWriteDifferentExpirationTimes() {
+		$response = $this->Controller->response;
+
+		$this->Cookie->write('Tracking.1', 'something', false, '2 seconds');
+
+		// Emulate the next request from the client.
+		$cookies = $response->cookie();
+		$_COOKIE = $this->_rebuildCookie($cookies);
+		$this->Cookie->startup($this->Controller);
+		$this->assertEquals('something', $this->Cookie->read('Tracking.1'));
+
+		// Write another entry
+		$this->Cookie->write('Tracking.2', 'something else', false, '5 seconds');
+
+		// Emulate another request from the client.
+		$cookies = $response->cookie() + $cookies; // The previous cookie should be merged.
+		$_COOKIE = $this->_rebuildCookie($cookies);
+		$this->Cookie->startup($this->Controller);
+
+		$this->assertEquals('something', $this->Cookie->read('Tracking.1'));
+		$this->assertEquals('something else', $this->Cookie->read('Tracking.2'));
+
+		sleep(2);
```

Comment:
It is waiting for the expiration time to pass. See `$this->assertNull($this->Cookie->read('Tracking.1'));` below.